internal/services/cdn/cdn_frontdoor_helpers.go

```diff
@@ -339,76 +383,265 @@ func updateRouteAssociations(d *pluginsdk.ResourceData, meta interface{}, routeI
+	return nil
+}
+
+// Validates that the CDN FrontDoor Custom Domain can be associated with the CDN FrontDoor Route
+func validateCustomDomainRoutes(input *[]parse.FrontDoorRouteId, customDomainID *parse.FrontDoorCustomDomainId) error {
+	if len(*input) == 0 || input == nil {
+		return nil
 	}
 
-	// Make sure the resource is referencing all of the custom domains that are associated with the route...
-	missingDomains := make([]string, 0)
+	// check for duplicates...
+	if err := routeSliceHasDuplicates(input, "CDN FrontDoor Route"); err != nil {
+		return err
+	}
 
-	for _, v := range routeCustomDomains {
-		// If this was updated by the portal, it lowercases to resource ID...
-		customDomain, err := parse.FrontDoorCustomDomainIDInsensitively(v.(string))
-		if err != nil {
-			return fmt.Errorf("unable to parse %q: %+v", v.(string), err)
+	for i, route := range *input {
+		// the route and custom domain profiles must match...
+		if customDomainID.ProfileName != route.ProfileName {
+			return fmt.Errorf("the CDN FrontDoor Custom Domain(Name: %q, Profile: %q) and the CDN FrontDoor Route(Name: %q, Profile: %q) must belong to the same CDN FrontDoor Profile", customDomainID.CustomDomainName, customDomainID.ProfileName, route.RouteName, route.ProfileName)
 		}
 
-		if !sliceContainsString(resourceCustomDomains, customDomain.ID()) {
-			missingDomains = append(missingDomains, fmt.Sprintf("%q", customDomain.ID()))
+		// validate all routes are using the same endpoint because a custom domain can not
+		// be associated with routes that target two different endpoints...
+		for t, v := range *input {
+			if i == t {
+				continue
+			}
+
+			if route.AfdEndpointName != v.AfdEndpointName {
+				return fmt.Errorf("the CDN FrontDoor Route(Name: %q) and CDN FrontDoor Route(Name: %q) do not reference the same CDN FrontDoor Endpoint(Name: %q). All CDN FrontDoor Routes must reference the same CDN FrontDoor Endpoint %q to associate the CDN FrontDoor Custom Domain(Name: %q) with more than one CDN FrontDoor Route", route.RouteName, v.RouteName, route.AfdEndpointName, route.AfdEndpointName, customDomainID.CustomDomainName)
+			}
 		}
 	}
 
-	if len(missingDomains) > 0 {
-		return fmt.Errorf("does not contain all of the CDN Front Door Custom Domains that are associated with the CDN Front Door Route(Name: %q). Please add the following CDN Front Door Custom Domain(s) to your CDN Route Disable Link To Default Domain configuration: %s", routeName, strings.Join(missingDomains, ", "))
+	return nil
+}
+
+// Returns a verbose CDN FrontDoor Custom Domain parse error message
+func friendlyCustomDomainID(customDomain string) (*parse.FrontDoorCustomDomainId, error) {
+	if customDomain != "" {
+		if customDomainId, err := parse.FrontDoorCustomDomainIDInsensitively(customDomain); err != nil {
+			return nil, fmt.Errorf("unable to parse CDN FrontDoor Custom Domain(ID: %q): %+v", customDomain, err)
+		} else {
+			return customDomainId, nil
+		}
```

Comment:
do we need this function? also this can be case-sensitive since it's from the config

```suggestion
func friendlyCustomDomainID(customDomain string) (*parse.FrontDoorCustomDomainId, error) {
	if customDomain != "" {
		if customDomainId, err := parse.FrontDoorCustomDomainID(customDomain); err != nil {
			return nil, fmt.Errorf("unable to parse CDN FrontDoor Custom Domain(ID: %q): %+v", customDomain, err)
		} else {
			return customDomainId, nil
		}
```

Reply:
Fixed.